File: careercup/binarytreecolumnwise.py
class Node:
	def __init__(self, val):
		self.val = val
		self.left = None
		self.right = None

def distrowcolumn(tree, col, row):
	if tree is None:
		return
	if col not in weights:
		weights[col] = {}
	distrowcolumn(tree.left, col-1, row-1)
	weights[col][row] = tree.val
	distrowcolumn(tree.right, col-1, row+1)
	

root =  Node(1)
root.left = Node(2)
root.right = Node(3)
root.left.left = Node(4)
root.left.right = Node(5)
root.right.left = Node(6)
root.right.right = Node(7)
root.right.left.right = Node(8)

weights = {}
distrowcolumn(root, 0, 0)
for col in sorted(weights.keys()):
	for row in sorted(weights[col].keys()):
		print weights[col][row]

File: careercup/binarytreecolumnwise.c
/*

Given the root of a binary tree, print the nodes column wise and row wise.

..............6
............/....\
...........9......4
........../..\......\
.........5....1.....3
..........\........./
...........0.......7
The answer would be 5 9 0 6 1 4 7 3.

https://www.careercup.com/question?id=5709428993556480

*/

void printtree(struct node * tree) {
	if (!tree)
		return;
	struct node *curr = tree->left;
	struct node *prev = tree;
	if (!curr) {
		printf("%d ", prev->value);
		printtree(prev->right);
	}
	while (curr->left) {
		prev = curr;
		curr = curr->left;
	}
	printf("%d ", curr->value);
	printf("%d ", prev->value);
	printtree(prev->right);
	printtree(curr->right)
	
}

void printtree(struct node *tree) {
	if (!tree)
		return;
	printtree(tree->left);
	printf("%d ", tree->value);
	printtree(tree->right);
}

void printtreecolumnrow(struct node *tree, int line, int col) {
	struct node *curr = tree;
	while (curr->left) {
		curr->line = line;
		curr->col = col;
		curr = curr->left;
		line-=1;
		col-=1;
	}	
